### ISSUE TITLE
feat: implement service bindings + local mode `dev`

### ISSUE THREAD

**Issue by threepointone** [cloudflare/wrangler2#1040]
Service bindings (`[services]`) works with remote mode in wrangler dev, we should figure out how to do the same for local mode. This is part of the broader picture for multi worker, but filing this as a followup for https://github.com/cloudflare/wrangler2/pull/906 anyway.

**Comment by petebacondarwin** [comment]
This is really cool @chrstntdd. Thanks for the write up. I think what you are doing is pretty much the best you can manage right now. I like the way you use the `FetchCtx` to allow browser and Worker rendering.
I wasn't quite sure where the `API_ENDPOINT` identifier comes in. I don't see it in your code. I guess it is in your wrangler.tomls? It would be interesting to see that part of the set up too.

**Comment by chrstntdd** [comment]
Ah yeah, apologies for leaving that bit out. It's not too involved at the moment, but it goes like this:

```ts
// env.ts

// `ENV` is injected by the build scripts with process.env.NODE_ENV or equivalent.
export const API_ENDPOINT = ENV === "production" ? "https://api.<WORKER>.com" : "http://localhost:3001"
```

And the usage within the application like so:
```tsx
import { createResource } from "solid-js"

import { API_ENDPOINT } from "./config/env"
import { useFetch } from "./global-ctx"

export function App() {
  let fetch = useFetch()
  let [x] = createResource(`${API_ENDPOINT}/get-some-data`, async (key) => {
    let content = await fetch(key)
    return content
  })

  return (
    <section>
      <div>SSR data from worker:</div>
      <pre>{JSON.stringify(x(), null, 2)}</pre>
    </section>
  )
}
```

**Comment by threepointone** [comment]
Hey @chrstntdd, would you like to try the solution we just landed in `wrangler@beta`? You simply have to run `wrangler dev --local` on both workers in different terminal tabs, and they should automatically bind to each other.

**Comment by chrstntdd** [comment]
Hey @threepointone!
I upgraded to the beta release and it runs smoothly! Was able to cut down the `fetch` facade that gets passed to the client to only:
```ts
// ...
      fetch: async (req, opts) => {
        let $$ = $fetch.create({
          fetch: (...x) => env.api.fetch(...x),
        })

        return $$(req, opts)
      },
```
Many thanks for the fixture you put together here 😄

**Comment by Qxxxx** [comment]
Hi @threepointone, has this feature already merged into `wrangler 2.14.0`?
